File: qc-core/src/main/java/com/eager/questioncloud/core/domain/question/infrastructure/repository/QuestionRepository.java
package com.eager.questioncloud.core.domain.question.infrastructure.repository;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.question.common.QuestionFilter;
import com.eager.questioncloud.core.domain.question.dto.QuestionInformation;
import com.eager.questioncloud.core.domain.question.model.Question;
import java.util.List;

public interface QuestionRepository {
    int countByQuestionFilter(QuestionFilter questionFilter);

    List<QuestionInformation> getQuestionInformation(QuestionFilter questionFilter);

    QuestionInformation getQuestionInformation(Long questionId, Long userId);

    List<Question> getQuestionsByQuestionIds(List<Long> questionIds);

    Boolean isAvailable(Long questionId);

    Question findByQuestionIdAndCreatorId(Long questionId, Long creatorId);

    Question get(Long questionId);

    Question save(Question question);

    List<QuestionInformation> findByCreatorIdWithPaging(Long creatorId, PagingInformation pagingInformation);

    List<QuestionInformation> findByQuestionIdIn(List<Long> questionIds);

    int countByCreatorId(Long creatorId);

    void increaseQuestionCount(Long questionId);

    void deleteAllInBatch();
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/question/infrastructure/repository/QuestionRepository.kt
package com.eager.questioncloud.core.domain.question.infrastructure.repository

import com.eager.questioncloud.core.common.PagingInformation
import com.eager.questioncloud.core.domain.question.common.QuestionFilter
import com.eager.questioncloud.core.domain.question.dto.QuestionInformation
import com.eager.questioncloud.core.domain.question.model.Question

interface QuestionRepository {
    fun countByQuestionFilter(questionFilter: QuestionFilter): Int

    fun getQuestionInformation(questionFilter: QuestionFilter): List<QuestionInformation>

    fun getQuestionInformation(questionId: Long, userId: Long): QuestionInformation

    fun getQuestionsByQuestionIds(questionIds: List<Long>): List<Question>

    fun isAvailable(questionId: Long): Boolean

    fun findByQuestionIdAndCreatorId(questionId: Long, creatorId: Long): Question

    fun get(questionId: Long): Question

    fun save(question: Question): Question

    fun findByCreatorIdWithPaging(creatorId: Long, pagingInformation: PagingInformation): List<QuestionInformation>

    fun findByQuestionIdIn(questionIds: List<Long>): List<QuestionInformation>

    fun countByCreatorId(creatorId: Long): Int

    fun increaseQuestionCount(questionId: Long)

    fun deleteAllInBatch()
}
